pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -459,8 +462,14 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloud.MachineCreateDe
 		})
 	}
 
+	var instanceMarketOptions *ec2.InstanceMarketOptionsRequest
+	if config.IsSpotInstance != nil && *config.IsSpotInstance {
+		instanceMarketOptions = &ec2.InstanceMarketOptionsRequest{MarketType: aws.String("spot")}
```

Comment:
nit: Theres a const for "spot" in the ec2 package: 
```go 
const (
	// MarketTypeSpot is a MarketType enum value
	MarketTypeSpot = "spot"
)
```

Do we also want to allow for configuration of `BlockDurationMinutes` ?

Reply:
SGTM and I was grepping for `Spot` not for `spot` as that is what they used in their example: https://aws.amazon.com/blogs/compute/new-amazon-ec2-spot-pricing/

> Do we also want to allow for configuration of BlockDurationMinutes ?

I guess thats reasonable. If and where we actually usage that option later on is not something we have to settle on right now